### DISCUSSION TITLE
Облегчённая версия для установки на роутеры keenetic без usb

### DISCUSSION THREAD

**Discussion by ZongerX** [bol-van/zapret#879] [Q&A]
Собственно вопрос: есть ли какие нибудь максимально обрезанные форки для установки на роутер keenetic без использования внешних usb накопителей? 

Есть статья на хабре где обсуждается установка на keenetic ultra с использованием usb накопителя, но к сожалению я имею keenetic start, без usb интерфейса. В комментариях подсказали что по сути вся установка аналогичная, но без каких то доп пакетов для разных архитектур. 

Как я понимаю здесь в гите универсальная версия под все платформы, но при установке в keenetic это избыточно и не хватит памяти для манипуляций, возможно ли сделать форк под ключ для беспроблемной установки на keenetic?

**Comment by bol-van** [comment]
На прошивках с недостатком чего-либо (в том числе и места) я предлагаю все делать руками
без скриптов. закачиваете туда tpws , ему ничего не надо. для nfqws придется подтянуть модули ядра как-то.
и ручками iptables. iptables, кстати, тоже на кинетике нет без entware, вроде (у меня нет кинетика и вряд ли будет)
вообще это в дискуссии надо

**Comment by Anonym-tsk** [comment]
Есть такие варианты. На KN-1011 работает с entware на внутренней памяти.
https://github.com/Anonym-tsk/nfqws-keenetic
https://github.com/Anonym-tsk/tpws-keenetic

**Comment by xfg** [comment]
> На прошивках с недостатком чего-либо (в том числе и места) я предлагаю все делать руками без скриптов. закачиваете туда tpws , ему ничего не надо. для nfqws придется подтянуть модули ядра как-то. и ручками iptables. iptables, кстати, тоже на кинетике нет без entware, вроде (у меня нет кинетика и вряд ли будет) вообще это в дискуссии надо

@bol-van как в таком случае указать tpws применять дурение к QUIC, достаточно иметь конфиг и tpws его подхватит или здесь нужен nfqws ?

**Comment by bol-van** [comment]
tpws работает только с tcp. quic умеет дурить только nfqws

**Comment by ZongerX** [comment]
Нужно устанавливать только tpws или nfqws тоже? Или это две не связанных друг с другом технологии? Я так понимаю nfqws лучше и свежее? Его установить нет варианта?

**Comment by NewUse** [comment]
Сделал для openwrt:
https://github.com/bol-van/zapret/pull/413
влезает даже на 4Мб флеш с gui без ipv6, правда
